library/core/src/main/java/com/google/android/exoplayer2/source/ads/AdsMediaSource.java

```diff
@@ -332,7 +332,7 @@ private void maybeUpdateAdMediaSources() {
   private void maybeUpdateSourceInfo() {
     @Nullable Timeline contentTimeline = this.contentTimeline;
     if (adPlaybackState != null && contentTimeline != null) {
-      if (adPlaybackState.adGroupCount == 0) {
+      if (adPlaybackState.adGroupCount == 0 || contentTimeline.getPeriodCount() > 1) {
```

Comment:
This modification is not necessary here.
This is mainly for the initialization of the ad manager.

After the administrator is initialized and before the ad tag response is obtained, only`contentTimeline` can be used. 
After the ad tag response is obtained, we will combine `contentTimeline` and related ads to construct a new`SinglePeriodAdTimeline` to use.